ensemble_eeg/ensemble_edf.py

```diff
@@ -219,7 +219,25 @@ def write_edf_header(fd, header):
                 if not isinstance(val, bytes):
                     val = str(val).encode(encoding="ascii").ljust(size, b"\x20")
 
-                assert len(val) == size
+                if len(val) > size:
+                    try:
+                        # convert float to scientific expression
+                        val = float(val)
+                        if val >= 0:
+                            precision = 2
+                        else:
+                            precision = 1
+                        val = bytes(f"{val:.{precision}e}", encoding="ascii").ljust(
+                            size, b" "
+                        )
+                    except:
+                        raise AssertionError(
+                            f"{val} too long! Need to be shorter than {size} bytes."
+                        )
```

Comment:
But then which exact piece of code is supposed to fail? If `val = float(val)`:

```suggestion
                    try:
                        # convert float to scientific expression
                        val = float(val)
                    except ValueError as e:
                        raise AssertionError(
                            f"{val} too long! Need to be shorter than {size} bytes."
                        ) from e
                    if val >= 0:
                        precision = 2
                    else:
                        precision = 1
                    val = bytes(f"{val:.{precision}e}", encoding="ascii").ljust(
                        size, b" "
                    )
```

Reply:
The idea is that the "main error" is that the content is too long for the number of characters that are dedicated to it, BUT if it is a float it can be mitigated by using scientific expression. Else, the error is raised anyway.